### ISSUE TITLE
today get "none" version  from resp[BUG] 

### ISSUE THREAD

**Issue by exbein** [JoMingyu/google-play-scraper#111]
**google_play_scraper.VERSION**
 1.0.2

**Describe the bug**
can't get version ,title,description and etc

**Code**
this is resp from request
{
    'title': None,
    'description': None,
    'descriptionHTML': None,
    'summary': None,
    'summaryHTML': None,
    'installs': None,
    'minInstalls': None,
    'score': None,
    'ratings': None,
    'reviews': None,
    'histogram': [0, 0, 0, 0, 0],
    'price': None,
    'free': None,
    'currency': None,
    'sale': False,
    'saleTime': None,
    'originalPrice': None,
    'saleText': None,
    'offersIAP': False,
    'inAppProductPrice': None,
}

**Comment by 89z** [comment]
please provide example app

**Comment by exbein** [comment]
try this 'com.exnoa.misttraingirls' and 'com.tgc.sky.android'
i have tried some,all return like this resp:
{
'title': None,
'description': None,
'descriptionHTML': None,
'summary': None,
'summaryHTML': None,
'installs': None,
'minInstalls': None,
'score': None,
'ratings': None,
'reviews': None,
'histogram': [0, 0, 0, 0, 0],
'price': None,
'free': None,
'currency': None,
'sale': False,
'saleTime': None,
'originalPrice': None,
'saleText': None,
'offersIAP': False,
'inAppProductPrice': None,
}

**Comment by varlesh** [comment]
I confirm this too. Show only comments

**Comment by StrangeFate** [comment]
The app(com.exnoa.misttraingirls) is not showing up on the website(google play store PC version). is the app visible on your pc?
also, I got 404 error on the colab.

i'm not sure why 'com.tgc.sky.android' is not showing any app detail though...

**Comment by ba11b0y** [comment]
+1 
The package has stopped working today and gets `None` for all apps.
Here's a sample package name to try with `com.demoapp.connect`

@JoMingyu @yshalsager

**Comment by varlesh** [comment]
Another scraper worked for me:
https://github.com/danieliu/play-scraper

**Comment by Arshit007** [comment]
same error coming for me as well

**Comment by ba11b0y** [comment]
> Another scraper worked for me: https://github.com/danieliu/play-scraper

This works just fine. Although the response is slightly changed. Thanks @varlesh

**Comment by varlesh** [comment]
![image](https://user-images.githubusercontent.com/8083855/148347455-33ae0ae0-a818-4f52-8c09-cd73060ffe99.png)

**Comment by StrangeFate** [comment]
It might be off the topic since I don't really know how the requesting of app detail works, but am I the only one who gets a newer design of google play store with the url provided in the app detail result? 

When I get the package name from the url, it shows the old version of the google play store as in the picture below.

![GF3](https://user-images.githubusercontent.com/79512178/148347599-28fccc90-f4df-4d60-975f-20da57ae13d3.PNG)

Which is turned as newer design of the google play store when I use the url provided from app detail result as below

![GF2](https://user-images.githubusercontent.com/79512178/148347792-912aec3f-b3b5-43c5-9db7-25ee99d820da.PNG)

Maybe this was the problem of it?

**Comment by 89z** [comment]
I have an alternate tool that works:

~~~
PS C:\> googleplay -a com.exnoa.misttraingirls
Title: ミストトレインガールズA ～霧の世界の車窓から～
UploadDate: Dec 2, 2021
VersionString: 1.3.2
VersionCode: 111
NumDownloads: 83.801 K
Size: 60.593 MB
Offer: 0.00 USD

PS C:\> googleplay -a com.tgc.sky.android
POST https://android.clients.google.com/auth
GET https://android.clients.google.com/fdfe/details?doc=com.tgc.sky.android
Title: Sky: Children of the Light
UploadDate: Nov 3, 2021
VersionString: 0.15.5 (179644)
VersionCode: 179644
NumDownloads: 13.832 M
Size: 1.346 GB
Offer: 0.00 USD
~~~

https://github.com/89z/googleplay

**Comment by mrommel** [comment]
+1

**Comment by svennickel** [comment]
As long I can see exchanging `./constants/element.py` with following code fixes mainly all fields, except `editorsChoise` and `moreByDeveloper`.

USE AT YOUR OWN RISK!

```
from datetime import datetime

from google_play_scraper.utils.data_processors import unescape_text

from typing import Callable, List, Any, Optional

from google_play_scraper.constants.regex import Regex
from google_play_scraper.utils import nested_lookup


class ElementSpec:
    def __init__(
        self,
        ds_num: Optional[int],
        data_map: List[int],
        post_processor: Callable = None,
        fallback_value: Any = None,
    ):
        self.ds_num = ds_num
        self.data_map = data_map
        self.post_processor = post_processor
        self.fallback_value = fallback_value

    def extract_content(self, source: dict) -> Any:
        try:
            if self.ds_num is None:
                result = nested_lookup(source, self.data_map)
            else:
                result = nested_lookup(
                    source["ds:{}".format(self.ds_num)], self.data_map
                )

            if self.post_processor is not None:
                result = self.post_processor(result)
        except:
            if isinstance(self.fallback_value, ElementSpec):
                result = self.fallback_value.extract_content(source)
            else:
                result = self.fallback_value

        return result


class ElementSpecs:
    Detail = {
        "title": ElementSpec(5, [0, 0, 0]),
        "description": ElementSpec(5, [0, 10, 0, 1], unescape_text),
        "descriptionHTML": ElementSpec(5, [0, 10, 0, 1]),
        "summary": ElementSpec(5, [0, 10, 1, 1], unescape_text),
        "summaryHTML": ElementSpec(5, [0, 10, 1, 1]),
        "installs": ElementSpec(5, [0, 12, 9, 0]),
        "minInstalls": ElementSpec(
            5, [0, 12, 9, 0], lambda s: int(Regex.NOT_NUMBER.sub("", s)) if s else 0
        ),
        "score": ElementSpec(6, [0, 6, 0, 1]), # former (7,
        "ratings": ElementSpec( # former (7,
            6, [0, 6, 1], lambda container: sum([item[1] for item in container[1:]])
        ),
        "reviews": ElementSpec(6, [0, 6, 3, 1]),
        "histogram": ElementSpec(
            6,
            [0, 6, 1],
            lambda container: [
                container[1][1],
                container[2][1],
                container[3][1],
                container[4][1],
                container[5][1],
            ],
            [0, 0, 0, 0, 0],
        ),
        "price": ElementSpec( # former (4,
            3, [0, 2, 0, 0, 0, 1, 0, 0], lambda price: (price / 1000000) or 0
        ),
        "free": ElementSpec(3, [0, 2, 0, 0, 0, 1, 0, 0], lambda s: s == 0),
        "currency": ElementSpec(3, [0, 2, 0, 0, 0, 1, 0, 1]),
        "sale": ElementSpec(3, [0, 2, 0, 0, 0, 14, 0, 0], bool, False),
        "saleTime": ElementSpec(3, [0, 2, 0, 0, 0, 14, 0, 0]),
        "originalPrice": ElementSpec(
            3, [0, 2, 0, 0, 0, 1, 1, 0], lambda price: (price / 1000000) or 0
        ),
        "saleText": ElementSpec(3, [0, 2, 0, 0, 0, 14, 1]),
        "offersIAP": ElementSpec(5, [0, 12, 12, 0], bool, False),
        "inAppProductPrice": ElementSpec(5, [0, 12, 12, 0]),
        "size": ElementSpec(8, [0]), # former (3,
        "androidVersion": ElementSpec(8, [2], lambda s: s.split()[0]),
        "androidVersionText": ElementSpec(8, [2]),
        "developer": ElementSpec(5, [0, 12, 5, 1]), # former (6, )
        "developerId": ElementSpec(5, [0, 12, 5, 5, 4, 2], lambda s: s.split("id=")[1]),
        "developerEmail": ElementSpec(5, [0, 12, 5, 2, 0]),
        "developerWebsite": ElementSpec(5, [0, 12, 5, 3, 5, 2]),
        "developerAddress": ElementSpec(5, [0, 12, 5, 4, 0]),
        "privacyPolicy": ElementSpec(5, [0, 12, 7, 2]),
        "developerInternalID": ElementSpec(5, [0, 12, 5, 0, 0]),
        "genre": ElementSpec(5, [0, 12, 13, 0, 0]),
        "genreId": ElementSpec(5, [0, 12, 13, 0, 2]),
        "icon": ElementSpec(5, [0, 12, 1, 3, 2]),
        "headerImage": ElementSpec(5, [0, 12, 2, 3, 2]),
        "screenshots": ElementSpec(
            5, [0, 12, 0], lambda container: [item[3][2] for item in container], []
        ),
        "video": ElementSpec(5, [0, 12, 3, 0, 3, 2]),
        "videoImage": ElementSpec(5, [0, 12, 3, 1, 3, 2]),
        "contentRating": ElementSpec(5, [0, 12, 4, 0]),
        "contentRatingDescription": ElementSpec(5, [0, 12, 4, 2, 1]),
        "adSupported": ElementSpec(5, [0, 12, 14, 0], bool),
        "containsAds": ElementSpec(5, [0, 12, 14, 0], bool, False),
        "released": ElementSpec(5, [0, 12, 36]),
        "updated": ElementSpec(5, [0, 12, 8, 0]),
        "version": ElementSpec(8, [1]), # former (4, [1])
        "recentChanges": ElementSpec(5, [0, 12, 6, 1], unescape_text),
        "recentChangesHTML": ElementSpec(5, [0, 12, 6, 1]),
        # TODO
        "comments": ElementSpec(
            18, [0], lambda container: [item[4] for item in container], []
        ),
        # TODO
        "editorsChoice": ElementSpec(5, [0, 12, 15, 0], bool, False),
        "similarApps": ElementSpec( # former (8,
            7,
            [1, 1, 0, 0, 0],
            lambda container: [container[i][12][0] for i in range(0, len(container))],
        ),
        # TODO
        "moreByDeveloper": ElementSpec(
            9,
            [0, 1, 0, 0, 0],
            lambda container: [container[i][12][0] for i in range(0, len(container))],
        ),
    }

    Review = {
        "reviewId": ElementSpec(None, [0]),
        "userName": ElementSpec(None, [1, 0]),
        "userImage": ElementSpec(None, [1, 1, 3, 2]),
        "content": ElementSpec(None, [4]),
        "score": ElementSpec(None, [2]),
        "thumbsUpCount": ElementSpec(None, [6]),
        "reviewCreatedVersion": ElementSpec(None, [10]),
        "at": ElementSpec(None, [5, 0], lambda v: datetime.fromtimestamp(v)),
        "replyContent": ElementSpec(None, [7, 1]),
        "repliedAt": ElementSpec(None, [7, 2, 0], lambda v: datetime.fromtimestamp(v)),
    }

    Permission_Type = ElementSpec(None, [0])
    Permission_List = ElementSpec(
        None, [2], lambda container: sorted([item[1] for item in container])
    )

```

**Comment by JoMingyu** [comment]
It fixed at version 1.0.3. Sorry for being late. I'll set some notifications for issues. Big thanks to @mrommel .
